src/CrvInfoHelper.cc

```diff
@@ -0,0 +1,263 @@
+      art::Handle<CrvStepCollection> const& crvSteps,
+      art::Handle<MCTrajectoryCollection> const& mcTrajectories,
+      CrvHitInfoRecoCollection &recoInfo, CrvHitInfoMCCollection &MCInfo,
+      CrvSummaryReco &recoSummary, CrvSummaryMC &MCSummary,
+      CrvPlaneInfoMCCollection &MCInfoPlane, double crvPlaneY) {
+    GeomHandle<CosmicRayShield> CRS;
+    GeomHandle<DetectorSystem> tdet;
+    if(!crvCoincidences.isValid()) return;
+    size_t nClusters=crvCoincidences->size();
+    for(size_t i=0; i<nClusters; i++)
+    {
+      const CrvCoincidenceCluster &cluster = crvCoincidences->at(i);
+
+      //fill the Reco collection
+      recoInfo.emplace_back(
+          cluster.GetCrvSectorType(),
+          tdet->toDetector(cluster.GetAvgHitPos()),
+          cluster.GetStartTime(), cluster.GetEndTime(),
+          cluster.GetPEs(),
+          cluster.GetCrvRecoPulses().size(),
+          cluster.GetLayers().size(),
+          cluster.GetSlope());
+    }
+
+    if(!crvRecoPulses.isValid()) return;
+    size_t nRecoPulses=crvRecoPulses->size();
+    recoSummary.totalPEs=0;
+    std::set<CRSScintillatorBarIndex> counters;
+    for(size_t i=0; i<nRecoPulses; i++)
+    {
+      recoSummary.totalPEs+=crvRecoPulses->at(i).GetPEs();
+      counters.insert(crvRecoPulses->at(i).GetScintillatorBarIndex());
+    }
+    recoSummary.nHitCounters=counters.size();
+
+
+    //fill the MC collection
+    if(crvCoincidencesMC.isValid())
+    {
+      size_t nClustersMC=crvCoincidencesMC->size();
+      if(nClusters!=nClustersMC) std::cout<<"The number of MC and reco CRV coincidence clusters does not match!"<<std::endl;
+      for(size_t i=0; i<nClustersMC; i++)
+      {
+        const CrvCoincidenceClusterMC &clusterMC = crvCoincidencesMC->at(i);
+        if(clusterMC.HasMCInfo())
+        {
+          const art::Ptr<SimParticle> &simParticle = clusterMC.GetMostLikelySimParticle();
+          const art::Ptr<SimParticle> &primaryParticle = FindPrimaryParticle(simParticle);
+          const art::Ptr<SimParticle> &parentParticle  = FindParentParticle(simParticle);
+          const art::Ptr<SimParticle> &gparentParticle = FindGParentParticle(simParticle);
+          MCInfo.emplace_back(
+              true,
+              simParticle->pdgId(),
+              primaryParticle->pdgId(),
+              primaryParticle->startMomentum().e() - primaryParticle->startMomentum().m(),
+              tdet->toDetector(primaryParticle->startPosition()),
+              parentParticle->pdgId(),
+              parentParticle->startMomentum().e()  - parentParticle->startMomentum().m(),
+              tdet->toDetector(parentParticle->startPosition()),
+              gparentParticle->pdgId(),
+              gparentParticle->startMomentum().e() - gparentParticle->startMomentum().m(),
+              tdet->toDetector(gparentParticle->startPosition()),
+              tdet->toDetector(clusterMC.GetEarliestHitPos()),
+              clusterMC.GetEarliestHitTime(),
```

Comment:
Should be replaced by GetAvgHitPos() and GetAvgHitTime()

Reply:
will do